File: src/plumbing_pub_sub/src/demo4_sub_person.cpp
#include "ros/ros.h"
#include "plumbing_pub_sub/Person.h"

/*
   订阅方：订阅数据
        1.包含头文件
        #include "plumbing_pub_sub/Person.h"
        2.初始化ros节点
        3.创建节点句柄
        4.创建订阅者对象
        5.编写订阅者逻辑，接收数据
        6.调用spin()函数
*/
void doPerson(const plumbing_pub_sub::Person::ConstPtr& person)
{
    ROS_INFO("订阅者的信息：%s,%d,%.3f",person->name.c_str(),person->age,person->height);
}
int main(int argc, char *argv[])
{
    setlocale(LC_ALL,"");
    ROS_INFO("订阅方实现");
    ros::init(argc,argv,"XianRen");
    ros::NodeHandle nh;
    ros::Subscriber sub =nh.subscribe("liaotian",10,doPerson);
    ros::spin();
    return 0;
}

File: src/plumbing_pub_sub/scripts/demo4_sub_person.py
#!/usr/bin/env python
import rospy
from plumbing_pub_sub.msg import Person

"""
        订阅流程:订阅人消息
            1.导包
            2.初始化ros节点
            3.创建订阅者对象
            4.回调函数处理数据
            5.spin()
"""

def doPerson (p):
    rospy.loginfo("小伙子的数据：%s,%d,%.2f",p.name,p.age,p.height)
    
if __name__=="__main__":

    rospy.init_node("DaYe")

    sub = rospy.Subscriber("jiaoSheTou",Person,doPerson,queue_size=10)

    rospy.spin()
    pass